lib/puppet/pops/types/types.rb

```diff
@@ -1307,8 +1524,9 @@ class PVariantType < PAnyType
 
       attr_reader :types
 
+      # @param types [Array[PAnyType]] the variants
       def initialize(types)
-        @types = types.freeze
+        @types = types.uniq.freeze
```

Comment:
Other optimizations here would be to reduce the set based on assignability. e.g:
`Variant[Integer, Float, Any]`, that is just Variant[Any], or infact, Any :-)